crates/wasmtime/src/module.rs

```diff
@@ -1042,6 +1042,34 @@ impl Module {
         self.inner.memory_images()?;
         Ok(())
     }
+
+    /// Get the map from `.text` section offsets to Wasm binary offsets for this
+    /// module.
+    ///
+    /// Each entry is a (`.text` section offset, Wasm binary offset) pair.
+    ///
+    /// Entries are yielded in order of `.text` section offset.
+    pub fn address_map<'a>(&'a self) -> impl Iterator<Item = (u32, Option<u32>)> + 'a {
+        wasmtime_environ::iterate_address_map(self.code_object().code_memory().address_map_data())
+            .expect("should have a valid address map section")
```

Comment:
I think this'll want to handle `None` and return an empty iterator?

Reply:
I guess the address map data is an empty slice when we've been configured not to have address maps? I was assuming that the only way it wouldn't be the correct format was if we messed up plumbing data around.